server/server.js

```diff
@@ -1644,13 +1646,20 @@ async function afterLogin(socket, user) {
 
     await StatusPage.sendStatusPageList(io, socket);
 
-    for (let monitorID in monitorList) {
-        await sendHeartbeatList(socket, monitorID);
+    // Create an array to store the combined promises for both sendHeartbeatList and sendStats
+    const monitorPromises = [];
+    for (let monitorID in monitorList.list) {
+        // Combine both sendHeartbeatList and sendStats for each monitor into a single Promise
+        monitorPromises.push(
+            Promise.all([
```

Comment:
I don't think these two need to be send in serial.
Have you checked what happens if you don't?

Reply:
@CommanderStorm 
understand but it makes a huge difference on loading time. I already test this scenario & make a decision.

if you call one by one for all monitor. it take time.

It's core part of changes while loading page. then next preload data.

can you explain why you think not required to send in serial?

Reply:
True, my logic was off on this one.

=> Let's just remove the inner `Promise.all` and just give all promises to the outer one.

Reply:
@CommanderStorm 
moved to outer one. thank you for suggestion